proxy/src/control/cache.rs

```diff
@@ -162,13 +187,20 @@ where
                 // If the new value isn't equal to the old value, overwrite
                 // the old value.
                 Some(new_value) =>  {
+                    // TODO: ideally, we wouldn't clone here, but we can't
+                    // borrow the value from `self.inner` after inserting it,
+                    // as `self.inner` is already borrowed mutably by `retain`.
+                    // It would be nice to figure this out.
                     let _ = mem::replace(value, new_value.clone());
```

Comment:
This is the one remaining `clone` call in this module. I couldn't figure out how to remove it easily. @briansmith, if you have any insight it would be greatly appreciated.

Reply:
Isn't `update_intersection()` dead code now? It seems like it was only used in this module and now its uses have been removed, so I think we can just delete it.

Reply:
Oh, I think you're right --- I mistakenly assumed it was part of the cache's API surface but now I realise it was only used internally. 

In that case, how do you feel about renaming `update_union` to just `update`, since we no longer have to distinguish the two operations?

Reply:
That's fair. I'll just remove the unused function, then.